share/functions/path_helper.fish

```diff
@@ -0,0 +1,22 @@
+# Reimplement the macOS /usr/libexec/path_helper executable for fish;
+# see __fish_construct_path in config.fish and
+# https://opensource.apple.com/source/shell_cmds/shell_cmds-203/path_helper/path_helper.c.auto.html .
+
+function __quote -d "single-quote arguments, escaping characters as necessary"
+    for x in $argv
+        set -l quoted (string replace -a -r '([\\\\\'])' '\\\\\\\\$1' $x)
```

Comment:
Not exactly sure why I need to triple-escape the `\` in `\\\\\\\\$1`; need to escape once because it's in a single-quoted string, another because it's a replacement string for a regexp, but I can't think of a reason for a third escape.